Improved Android compatibility

Burtan opened the pull request:
Hey, this fixes will not change anything on JDK8 desktop use-cases but fix using missing classes on Android and headless runtimes.

hoijui commented on modules/core/src/main/java/com/illposed/osc/LibraryInfo.java:
```diff
@@ -286,6 +288,19 @@ public static boolean isAndroidApi23AndLower() {
+	 */
+	public static boolean isDesktop() {
+		try {
+			Class.forName("java.awt.color");
```
should it not be `Color` instead of `color`?

Burtan replied:
You are probably right ;-)

hoijui commented on modules/core/src/main/java/com/illposed/osc/LibraryInfo.java:
```diff
@@ -286,6 +288,19 @@ public static boolean isAndroidApi23AndLower() {
+			Class.forName("java.awt.color");
+			return true;
+		} catch (ClassNotFoundException ignore) {
+			//my class isn't there!
```
remove or change to something like "This class is part of JDK, so we are not on JDK -> assuming we are on Android"?

Burtan replied:
You probably meant the StandardProtocolFamily part? I did some renaming and changed the description. I hope you like it better.

hoijui replied:
ah.. I guess that is also a good change, though I would have been fine with it before too. ;-)
.. I was concerned about the `isDesktop()` method. Many developpers nowadays only know Android or only know the JDK world, and the method should be named in a way, and documented in a way that makes it understandable for both of these.

I say this, because I know very little about Android, and only since a year or two, know that is does not have `java.awt.Color`, for example. ;-)
.. and I like to make code save for people as ignorant as I am (was) myself.

I would probably revert the boolean value of the function, and call it `isAndroid()` instead, because JDK can be used on headless server, on embedded systems and desktops.

Burtan replied:
I made the function's naming more generic and put the Android stuff into its descriptions.

hoijui replied:
uff, much better then what I though of, great! :-)

hoijui commented:
thank you, the code looks pretty much perfect now! :-)
I made two small commits on top of this in the [android-support branch](https://github.com/hoijui/JavaOSC/commits/android-support).
If you could cleanup the history with `rebase -i` to remove all the adding and removing, changing and changing back of things, to come up with less noise in the history, and only your name for the changes, it is ready for merge! :-)

Burtan commented:
git rebase -i seems to be more complicate than I thought, never used it yet. Gotta look into it.

hoijui commented:
ouh cool!! :D
it is a very powerful thing, and you will benefit a lot from knowing it!
If you really don't want to do it, I can do it for you (retaining your author and committer name and email of course), it is a daily thing for me/easy, but.. I can recommend to learn it! .. if you have time.

Burtan commented:
Give me some days, if I didn't succeed then, you can still do it yourself :D

Burtan commented:
Ok, I think i succeeded :D. I pulled your changes from the android-support branch, rebased my master branch on yours (no conflict), merged all of my commits into one and changed the authoring to reference to my github account.

hoijui commented:
you did! great! :D
thank you, hope to see you again and.. good night!

ManuDev9 commented:
Hello my friends,
this is a Java workaround from the application perspective. No need to pick weird snapshots:
```
            OSCSerializerAndParserBuilder serializer = new OSCSerializerAndParserBuilder();
            serializer.setUsingDefaultHandlers(false);
            List<ArgumentHandler> defaultParserTypes = Activator.createSerializerTypes();
            defaultParserTypes.remove(16);
            char typeChar = 'a';
            for (ArgumentHandler argumentHandler:defaultParserTypes) {
                serializer.registerArgumentHandler(argumentHandler, typeChar);
                typeChar++;
            }
            mOscPortOut = new OSCPortOut(serializer,
                    new InetSocketAddress(AppData.getOscRemoteAddress(), AppData.getOscPortOutNumber()));
```
Enjoy
